src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/MicrosoftQualityGuidelinesAnalyzersResources.resx

```diff
@@ -252,4 +252,13 @@
   <data name="AvoidPropertySelfAssignmentMessage" xml:space="preserve">
     <value>The property {0} should not be assigned to itself.</value>
   </data>
-</root>
\ No newline at end of file
+  <data name="ReferencingAndReassigningObjectInSameStatementMessage" xml:space="preserve">
+    <value>Object '{0}' is referenced and reassigned in the same statement. You are at risk of referring to unintended object.</value>
```

Comment:
I would also change the message to `Symbol '{0}' and its member '{1}' are both assigned in the same statement. You are at risk of assigning the member of an unintended object.`

Reply:
And also the title along the same lines...

Reply:
Yeah, that's better way of explaining it